### PR TITLE
Master "fix make generate" PR

### PR DESCRIPTION
**Pull request by peter-edge**
This first runs `glide update && make generate` as `make generate` is out of date.

Next, I added `make generatenodiff` to `make lint` so we can verify this on CI.

This produced bugs https://github.com/yarpc/yarpc-go/issues/815, https://github.com/yarpc/yarpc-go/issues/816, https://github.com/yarpc/yarpc-go/issues/817, which are fixed here as well.

I also did a bunch more cleanup.

This PR is more data for why Docker would be so nice top to bottom.

### CODE REVIEW

**Comment by breerly** [comment]
(Make sure Travis is happy)

**Comment by peter-edge** [comment]
Travis passed, #821 only affects exactly one of the five builds, just re-running that pre-merge

File: Makefile [hunk 1]
```diff
@@ -67,51 +63,90 @@ CI_DOCKER_CACHE_DIR := $(CI_CACHE_DIR)/docker
...
+$(BIN)/docker-compose:
+	mkdir -p $(BIN)
+	curl -L https://github.com/docker/compose/releases/download/$(DOCKER_COMPOSE_VERSION)/docker-compose-$(shell uname -s)-$(shell uname -m) > $(BIN)/docker-compose
+	chmod +x $(BIN)/docker-compose
+
+$(BIN)/glide:
+	mkdir -p $(BIN)
+	mkdir -p $(TMP)/glide
+	curl -L https://github.com/Masterminds/glide/releases/download/v$(GLIDE_VERSION)/glide-v$(GLIDE_VERSION)-$(shell uname -s)-amd64.tar.gz > $(TMP)/glide/glide.tar.gz
+	tar -C $(TMP)/glide -xzf $(TMP)/glide/glide.tar.gz
```
Comment (abhinav) [inline Makefile]: Don't need to change, just FYI: `cd`s in a single instruction in a Makefile don't propagate to the next so you can also do,

    cd $(TMP)/glide; tar xzf glide.tar.gz

File: Makefile [hunk 2]
```diff
@@ -67,51 +63,90 @@ CI_DOCKER_CACHE_DIR := $(CI_CACHE_DIR)/docker
...
+	rm -rf $(TMP)/thrift
+
+$(BIN)/thriftrw-plugin-yarpc: ./encoding/thrift/thriftrw-plugin-yarpc/*.go
+	mkdir -p $(BIN)
+	go build -o $(BIN)/thriftrw-plugin-yarpc ./encoding/thrift/thriftrw-plugin-yarpc
+
+DOCKER_COMPOSE = $(BIN)/docker-compose
+GLIDE = $(BIN)/glide
+THRIFT = $(BIN)/thrift
+BINS = $(DOCKER_COMPOSE) $(GLIDE) $(THRIFT) $(BIN)/thriftrw-plugin-yarpc
```
Comment (abhinav) [inline Makefile]: Little nit: Why not declare `GLIDE`, `DOCKER_COMPOSE`, etc. before the rules to build them and use `$(GLIDE)` as the rule target?

```
GLIDE = $(BIN)/glide

$(GLIDE):
    mkdir -p $(BIN)
    ...
```

File: Makefile [hunk 3]
```diff
@@ -165,8 +201,8 @@ errcheck:
 	@errcheck $(ERRCHECK_FLAGS) $(PACKAGES) 2>&1 | $(FILTER_LINT) | $(FILTER_ERRCHECK) > $(ERRCHECK_LOG) || true
 	@[ ! -s "$(ERRCHECK_LOG)" ] || (echo "errcheck failed:" | cat - $(ERRCHECK_LOG) && false)
 
-.PHONY: verify_version
-verify_version:
+.PHONY: verifyversion
+verifyversion:
```
Comment (abhinav) [inline Makefile]: Nit: update the instructions in RELEASING.md to reflect this change

File: Makefile [hunk 4]
```diff
@@ -222,32 +236,24 @@ examples:
...
-crossdock: $(_BIN_DIR)/docker-compose
-	PATH=$(_BIN_DIR):$$PATH docker-compose kill go
-	PATH=$(_BIN_DIR):$$PATH docker-compose rm -f go
-	PATH=$(_BIN_DIR):$$PATH docker-compose build go
-	PATH=$(_BIN_DIR):$$PATH docker-compose run crossdock
+crossdock: $(DOCKER_COMPOSE)
+	PATH=$(BIN):$$PATH docker-compose kill go
+	PATH=$(BIN):$$PATH docker-compose rm -f go
+	PATH=$(BIN):$$PATH docker-compose build go
+	PATH=$(BIN):$$PATH docker-compose run crossdock
```
Comment (abhinav) [inline Makefile]: If we have the full path to the executable, do we really need to prepend the directory to the PATH? Can't we do,

    $(DOCKER_COMPOSE) kill go

**Pull request merged**